src/ncp/ncp_spinel.cpp

```diff
@@ -266,6 +425,180 @@ spinel_tid_t NcpSpinel::GetNextTid(void)
     return tid;
 }
 
+otError NcpSpinel::SetProperty(spinel_prop_key_t aKey, const EncodingFunc &aEncodingFunc)
+{
+    otError      error  = OT_ERROR_NONE;
+    spinel_tid_t tid    = GetNextTid();
+    uint8_t      header = SPINEL_HEADER_FLAG | SPINEL_HEADER_IID(mIid) | tid;
+
+    VerifyOrExit(tid != 0, error = OT_ERROR_BUSY);
+    SuccessOrExit(error = mEncoder.BeginFrame(header, SPINEL_CMD_PROP_VALUE_SET, aKey));
+    SuccessOrExit(error = aEncodingFunc());
+    SuccessOrExit(error = mEncoder.EndFrame());
+    SuccessOrExit(error = SendEncodedFrame());
+
+    mWaitingKeyTable[tid] = aKey;
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        FreeTid(tid);
+    }
+    return error;
+}
+
+otError NcpSpinel::SendEncodedFrame(void)
+{
+    otError  error = OT_ERROR_NONE;
+    uint8_t  frame[kTxBufferSize];
+    uint16_t frameLength;
+
+    SuccessOrExit(error = mNcpBuffer.OutFrameBegin());
+    frameLength = mNcpBuffer.OutFrameGetLength();
+    VerifyOrExit(mNcpBuffer.OutFrameRead(frameLength, frame) == frameLength, error = OT_ERROR_FAILED);
+    SuccessOrExit(error = mSpinelDriver->GetSpinelInterface()->SendFrame(frame, frameLength));
+
+exit:
+    error = mNcpBuffer.OutFrameRemove();
+    return error;
+}
+
+void NcpSpinel::GetFlagsFromSecurityPolicy(const otSecurityPolicy *aSecurityPolicy,
+                                           uint8_t                *aFlags,
+                                           uint8_t                 aFlagsLength)
+{
+    static constexpr uint8_t kObtainNetworkKeyMask        = 1 << 7;
+    static constexpr uint8_t kNativeCommissioningMask     = 1 << 6;
+    static constexpr uint8_t kRoutersMask                 = 1 << 5;
+    static constexpr uint8_t kExternalCommissioningMask   = 1 << 4;
+    static constexpr uint8_t kCommercialCommissioningMask = 1 << 2;
+    static constexpr uint8_t kAutonomousEnrollmentMask    = 1 << 1;
+    static constexpr uint8_t kNetworkKeyProvisioningMask  = 1 << 0;
+    static constexpr uint8_t kTobleLinkMask               = 1 << 7;
+    static constexpr uint8_t kNonCcmRoutersMask           = 1 << 6;
```

Comment:
Probably document the byte index of the masks because there are same mask values in different bytes representing different meanings.